app/seed-data/emissions_factors/scripts/EFDB_2006_IPCC_guidelines_Stationary_Energy.py

```diff
@@ -0,0 +1,631 @@
+        },
+        'Diesel Oil': {
+            'value': 840, 
+            'units': 'kg/m3',
+            'reference': 'EF_GHG_protocol'
+        }, 
+        'Residual Fuel Oil': {
+            'value': 940, 
+            'units': 'kg/m3',
+            'reference': 'EF_GHG_protocol'
+        }, 
+        'Natural Gas': {
+            'value': 0.7, 
+            'units': 'kg/m3',
+            'reference': 'EF_GHG_protocol'
+        },
+        'Charcoal': {
+            'value': 208, 
+            'units': 'kg/m3',
+            'reference': 'Aqua_Calc_tool'
+        },
+        'Sub-Bituminous Coal': {
+            'value': 1346, 
+            'units': 'kg/m3',
+            'reference': 'Aqua_Calc_tool'
+        }, 
+        'Liquefied Petroleum Gases': {
+            'value': 540, 
+            'units': 'kg/m3',
+            'reference': 'EF_GHG_protocol'
+        }, 
+        'Coking Coal': {
+            'value': 829.76, 
+            'units': 'kg/m3',
+            'reference': 'Aqua_Calc_tool'
+        },
+        'Coke Oven Coke and Lignite Coke': {
+            'value': 400, 
+            'units': 'kg/m3',
+            'reference': 'Aqua_Calc_tool'
+        }, 
+        'Naphtha': {
+            'value': 770, 
+            'units': 'kg/m3',
+            'reference': 'EF_GHG_protocol'
+        }, 
+        'Coke Oven Gas': {
+            'value': 0.545, 
+            'units': 'kg/m3',
+            'reference': 'engineering_tool_box'
+        },
+        'Natural Gas Liquids\n(NGLs)': {
+            'value': 500, 
+            'units': 'kg/m3',
+            'reference': 'Aqua_Calc_tool'
+        },
+        'Jet Kerosene': {
+            'value': 790, 
+            'units': 'kg/m3',
+            'reference': 'EF_GHG_protocol'
+        }, 
+        'Bitumen': {
+            'value': 1346, 
+            'units': 'kg/m3',
+            'reference': 'Aqua_Calc_tool'
+        }
+    }
+
+    # fill density values in EF_df
+    for index, row in EF_df.iterrows():
+        if pd.isna(row['density_value']):
+            fuel = row['fuel']
+            if fuel in densities_dic.keys():
+                # Fill the missing density_value and density_units using the dictionary
+                EF_df.at[index, 'density_value'] = densities_dic[fuel]['value']
+                EF_df.at[index, 'density_units'] = densities_dic[fuel]['units']
+
+    # Conversion process
+    def convert_units(df, unit_col, value_col, from_unit, to_unit, conversion_factor):
+        # Filter the DataFrame based on the from_unit
+        filtered_df = df[df[unit_col] == from_unit]
+        # conversion
+        filtered_df[unit_col] = to_unit
+        filtered_df[value_col] *= conversion_factor
+        df.update(filtered_df)
+
+    # Define the conversions and their respective factors
+    conversions = [
+        ('tC/TJ', 'kg/TJ', 44/12),
+        ('gC/MJ', 'g/MJ', 44/12),
+        ('g/MJ', 'kg/TJ', 1),
+        ('kg/GJ', 'kg/TJ', 1000)
+    ]
```

Comment:
Might be good to store these in a constant at the top of the file?
So they can be edited and expanded more easily in the future?

Reply:
It's fine if it makes more sense for you to have it where it is in the code though :)